### PR TITLE
fix(streaming): add vnode check for state table `write_chunk`

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR adds the vnode checking for the `write_chunk` interface of the state table. So that we can check whether the distribution of records written with `write_chunk` is correct.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- #5151 
- #5344 
- #5322

### CODE REVIEW

File: risedev.yml [hunk 1]
```diff
@@ -20,7 +20,6 @@ risedev:
       enable-dashboard-v2: false
       unsafe-disable-recovery: true
     - use: compute-node
-      enable-in-memory-kv-state-backend: true
```
Comment (BugenZhao) [inline risedev.yml:23]: This line is added by accident. 🤣

File: src/storage/src/table/mod.rs [hunk 2]
```diff
@@ -108,24 +108,38 @@ pub trait TableIter: Send {
...
         DEFAULT_VNODE
     } else {
-        row.hash_by_indices(indices, &CRC32FastBuilder {})
+        row.hash_by_indices(indices, &Crc32FastBuilder {})
             .to_vnode()
     };
 
     tracing::trace!(target: "events::storage::storage_table", "compute vnode: {:?} key {:?} => {}", row, indices, vnode);
 
-    // FIXME: temporary workaround for local agg, may not needed after we have a vnode builder
```
Comment (BugenZhao) [inline src/storage/src/table/mod.rs:122]: As we already introduced `vnode_index` in #5079, this is unnecessary now.

**Review by wcy-fdu — APPROVED** [review]
LGTM

**Review by fuyufjh — APPROVED** [review]

File: src/storage/src/table/mod.rs (hunk 2 above)
Comment (BugenZhao) [reply src/storage/src/table/mod.rs:122]: Oops. Looks like not all local agg goes to `vnode_in_col` branch, so this makes e2e fail. 🥵

**Review by richardchien — APPROVED** [review]

File: src/storage/src/table/mod.rs (hunk 2 above)
Comment (BugenZhao) [reply src/storage/src/table/mod.rs:122]: I've opened an issue at #5420.

**Pull request merged**